src/Layers/xrRender_R2/r2_R_lights.cpp

```diff
@@ -129,14 +130,14 @@ void CRender::render_lights(light_Package& LP)
         }
     };
 
-    const auto& flush_lights = [&]()
+    const auto& flush_lights = [] (xr_vector<task_data_t>& lights_queue)
     {
         for (const auto& [L, task, batch_id] : lights_queue)
         {
             VERIFY(task);
             TaskScheduler->Wait(*task);
 
-            auto& dsgraph = get_context(batch_id);
+            auto& dsgraph = RImplementation.get_context(batch_id);
```

Comment:
This is still look weird because capturing `this` and then using CRender functions directly looks more natural.

And it's not proven that inlining would make a difference. Can we measure it?

Reply:
Well, at least for gcc 13.2, it doesn't seem to make any difference.
https://godbolt.org/z/8WafxGxoG
https://godbolt.org/z/M56ns3afq

I'll return the capture

Reply:
But still, don't just return original code which captures everything. It's better to capture `this`, that makes lambda-object smaller (if the compiler doesn't automatically deduce that it needs to capture `this`)

Reply:
Ah, you did it already :)